fawltydeps/types.py

```diff
@@ -83,7 +83,33 @@ def __post_init__(self) -> None:
         assert self.path.is_file()  # sanity check
 
 
-Source = Union[CodeSource, DepsSource]
+@dataclass(frozen=True, eq=True, order=True)
+class PyEnvSource:
+    """A source to be used for looking up installed Python packages.
+
+    This corresponds to the lib/pythonX.Y/site-packages directory in a
+    system-wide Python installation (under /usr, /usr/local, or similar),
+    in a virtualenv, in a poetry2nix environment, or a similar mechanism that
+    uses this layout. Alternatively, it can be a __pypackages__/X.Y/lib
+    directory within a project that uses PEP582.
+    """
+
+    path: Path
+
+    def __post_init__(self) -> None:
+        assert self.path.is_dir()  # sanity check
```

Comment:
Do you mean that we should rename the class members themselves (`.path` -> `.dir` or `.file`)?

I'm usually fond of `.path` since it also helps to remind me that this is a `Path` object. There is technically no problem with using `.dir` and `.file` as member names, but since they are also built-in functions in Python (`dir()` and `file()`), I'm usually a little hesitant to use them...

Otherwise I agree that we should be as clear as possible about this in the documentation.

Reply:
Improved the documentation of these members in 737e14ffa0531fbd404246c96de49b8c2fa99e82